crates/service-client/src/http.rs

```diff
@@ -153,13 +159,16 @@ impl HttpClient {
             Err(err) => return future::ready(Err(err.into())).right_future(),
         };
 
-        let client = match request.version() {
-            Version::HTTP_2 => &self.h2_client,
-            _ => &self.alpn_client,
+        let fut = match (
+            request.version(),
+            request.uri().scheme().expect("URI should be absolute"),
+        ) {
+            (Version::HTTP_2, scheme) if scheme == &Scheme::HTTP => {
+                self.h2c_prior_knowledge_client.request(request)
```

Comment:
Why can't we always use `h2c_prior_knowledge_client` if the version is `HTTP_2`?

Reply:
There's no need to create two https clients, we just use one here.

I think there's something here with the fact that we should work on our own pooling system in future, that among the other problems, can address those issues.

Reply:
Is this strictly required?

Reply:
it is required because of this change: https://github.com/restatedev/restate/pull/1900/files#diff-dcf1a0303314188fa354df1a0a0c2e4df9345d7e4911707b7042cfe6ae48f1c8R75

Reply:
Ah ok, I think I know understand. Thanks for the clarification.

So before, we were only using the ALPN enabled client if the user forced the usage of `HTTP1.1`, right? So when running on a server with https and `HTTP1.1` it wouldn't have worked w/o the explicit forcing. Now when encountering https, we'll always use the ALPN enabled client. Consequently, users will only have to force the usage of `HTTP1.1` if they are running a service on an http server w/o TLS using HTTP1.1, right?